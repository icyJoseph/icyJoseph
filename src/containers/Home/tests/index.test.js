import React from "react";
import { shallow } from "enzyme";
import Media from "react-media";
import { Home } from "../";

describe("Home", () => {
  const wrapper = shallow(<Home />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Mobile", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      configurable: true,
      value: query => {
        return {
          matches: query === "(max-width: 920px)",
          addListener: () => {},
          removeListener: () => {}
        };
      }
    });
  });

  afterAll(() => {
    delete window.matchMedia;
  });

  const nonSpy = jest.spyOn(Home.prototype, "renderGrid");
  const spy = jest.spyOn(Home.prototype, "renderMobile");
  const mobileWrapper = shallow(<Home />);
  it("renders mobile", () => {
    expect(mobileWrapper).toMatchSnapshot();
    mobileWrapper.render();
    expect(spy).toHaveBeenCalled();
    expect(nonSpy).toHaveBeenCalledTimes(0);
  });
});

describe("Desktop", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      configurable: true,
      value: query => {
        return {
          matches: query === "(min-width: 921px)",
          addListener: () => {},
          removeListener: () => {}
        };
      }
    });
  });

  afterAll(() => {
    delete window.matchMedia;
  });
  const spy = jest.spyOn(Home.prototype, "renderGrid");
  const nonSpy = jest.spyOn(Home.prototype, "renderMobile");
  const desktopWrapper = shallow(<Home />);
  it("renders desktop", () => {
    expect(desktopWrapper).toMatchSnapshot();
    desktopWrapper.render();
    expect(spy).toHaveBeenCalled();
    // The MobileView function gets called with no arguments
    // This is most likely a bug with react-media
    // To make sure it is not rendered we check for the length at Media[0]
    expect(nonSpy).toHaveBeenCalledTimes(1);
    expect(
      desktopWrapper
        .find(Media)
        .at(0)
        .render()
    ).toHaveLength(0);
  });
});
