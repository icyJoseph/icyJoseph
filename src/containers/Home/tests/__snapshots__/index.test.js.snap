// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop renders desktop 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Media
          defaultMatches={true}
          query={
                    Object {
                              "maxWidth": 920,
                            }
          }
          render={[Function]}
/>,
        <Media
          defaultMatches={true}
          query={
                    Object {
                              "minWidth": 921,
                            }
          }
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMatches": true,
          "query": Object {
            "maxWidth": 920,
          },
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMatches": true,
          "query": Object {
            "minWidth": 921,
          },
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Media
            defaultMatches={true}
            query={
                        Object {
                                    "maxWidth": 920,
                                  }
            }
            render={[Function]}
/>,
          <Media
            defaultMatches={true}
            query={
                        Object {
                                    "minWidth": 921,
                                  }
            }
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMatches": true,
            "query": Object {
              "maxWidth": 920,
            },
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMatches": true,
            "query": Object {
              "minWidth": 921,
            },
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Home renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Media
          defaultMatches={true}
          query={
                    Object {
                              "maxWidth": 920,
                            }
          }
          render={[Function]}
/>,
        <Media
          defaultMatches={true}
          query={
                    Object {
                              "minWidth": 921,
                            }
          }
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMatches": true,
          "query": Object {
            "maxWidth": 920,
          },
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMatches": true,
          "query": Object {
            "minWidth": 921,
          },
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Media
            defaultMatches={true}
            query={
                        Object {
                                    "maxWidth": 920,
                                  }
            }
            render={[Function]}
/>,
          <Media
            defaultMatches={true}
            query={
                        Object {
                                    "minWidth": 921,
                                  }
            }
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMatches": true,
            "query": Object {
              "maxWidth": 920,
            },
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMatches": true,
            "query": Object {
              "minWidth": 921,
            },
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Mobile renders mobile 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Media
          defaultMatches={true}
          query={
                    Object {
                              "maxWidth": 920,
                            }
          }
          render={[Function]}
/>,
        <Media
          defaultMatches={true}
          query={
                    Object {
                              "minWidth": 921,
                            }
          }
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMatches": true,
          "query": Object {
            "maxWidth": 920,
          },
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMatches": true,
          "query": Object {
            "minWidth": 921,
          },
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Media
            defaultMatches={true}
            query={
                        Object {
                                    "maxWidth": 920,
                                  }
            }
            render={[Function]}
/>,
          <Media
            defaultMatches={true}
            query={
                        Object {
                                    "minWidth": 921,
                                  }
            }
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMatches": true,
            "query": Object {
              "maxWidth": 920,
            },
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMatches": true,
            "query": Object {
              "minWidth": 921,
            },
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
